File: ui/goose2/src/shared/api/doctor.ts
import { invoke } from "@tauri-apps/api/core";

export type FixType = "command" | "bridge";

export interface DoctorCheck {
  id: string;
  label: string;
  status: "pass" | "warn" | "fail";
  message: string;
  fixUrl: string | null;
  fixCommand: string | null;
  fixType: FixType | null;
  path: string | null;
  bridgePath: string | null;
  rawOutput: string | null;
}

export interface DoctorReport {
  checks: DoctorCheck[];
}

export async function runDoctor(): Promise<DoctorReport> {
  return invoke("run_doctor");
}

export async function runDoctorFix(
  checkId: string,
  fixType: FixType,
): Promise<void> {
  return invoke("run_doctor_fix", { checkId, fixType });
}

File: ui/goose2/src-tauri/src/commands/doctor.rs
//! Tauri command wrappers for the doctor health-check system.

pub use doctor::{DoctorReport, FixType};

/// Run all health checks and return the report.
#[tauri::command]
pub async fn run_doctor() -> DoctorReport {
    doctor::run_checks().await
}

/// Run a fix command for a doctor check, identified by check ID and fix type.
#[tauri::command]
pub async fn run_doctor_fix(check_id: String, fix_type: FixType) -> Result<(), String> {
    doctor::execute_fix(check_id, fix_type).await
}
